packages/members-api/index.js

```diff
@@ -140,6 +124,29 @@ module.exports = function MembersApi({
         return encodeIdentityToken({sub: member.email});
     }
 
+    async function setMemberGeolocationFromIp(email, ip) {
+        if (!email || !ip) {
+            // TODO: feel like this should actually error
+            return null;
+        }
```

Comment:
The rest of the library is not throwing any errors outside of middleware which writes directly to the http response. Is that intentional?

Reply:
Kinda yeah - when we were first building this it was built in a much more "decoupled from Ghost" way - 

In an effort to simplify things the interface for all the "getters"/"setters" that are passed in was that they wouldn't error - they would either do the thing or return `null` - this made it easier to work with because we didn't have to handle errors, and we'd do a big try/catch in the handlers like you see - which by definition were "Unexpected/Internal Server" errors

I don't think that this pattern needs to continue now though - as we know that this code is inherently a part of Ghost now. Does that make sense?

Reply:
I changed this to throw relevant errors, the method is only used in `members-ssr` which can then handle it however it wants (in this case, ignore errors but log using `debug` so it can be investigated if needed)